File: mybatis/src/main/java/data/spring/mybatis/adapter/in/dto/ProductUpdateBatchRequest.java
package data.spring.mybatis.adapter.in.dto;

import data.spring.mybatis.application.service.command.ProductUpdateCommand;
import jakarta.validation.Valid;

import java.util.List;

public record ProductUpdateBatchRequest(
        List<@Valid ProductUpdateRequest> updateRequests
) {
    public List<ProductUpdateCommand> toCommands() {
        return this.updateRequests.stream()
                .map(request ->
                        ProductUpdateCommand.of(
                                request.productId(),
                                request.productName(),
                                request.price(),
                                request.quantity()
                ))
                .toList();
    }
}

File: mybatis/src/main/java/data/spring/mybatis/adapter/in/product/dto/ProductUpdateBatchRequest.kt
package data.spring.mybatis.adapter.`in`.product.dto

import data.spring.mybatis.application.service.product.command.ProductUpdateCommand
import jakarta.validation.Valid

data class ProductUpdateBatchRequest(
    @field:Valid
    val updateRequests: List<ProductUpdateRequest>
) {
    fun toCommands(): List<ProductUpdateCommand> {
        return this.updateRequests.stream()
            .map { request ->
                ProductUpdateCommand(
                    productId = request.productId,
                    productName = request.productName,
                    price = request.price,
                    quantity = request.quantity
                )
            }
            .toList()
    }
}
